src/realm/cluster.cpp

```diff
@@ -1653,24 +1653,68 @@ ref_type Cluster::typed_write(ref_type ref, _impl::ArrayWriterBase& out, const T
             else if (col_type == col_type_Mixed) {
                 const auto sz = leaf.size();
                 REALM_ASSERT(sz == 6);
-                // temporary disable mixed. in order to re-enable them in a separate PR
+
+                /*
+                 In order to know how Mixed stores things, we need to take in consideration this enum
+                 enum {
+                 payload_idx_type,
+                 payload_idx_int,
+                 payload_idx_pair,
+                 payload_idx_str,
+                 payload_idx_ref,
+                 payload_idx_key,
+                 payload_idx_size
+                 };
+                 Note:
+                 1. entry at index 0 is the composite array (the main information about the data is stored, it can
+                 either be a small int <32 bit or a ref to the array where the actual data is stored).
```

Comment:
Using the word 'ref' here is misleading. It is the offset into the relevant other array.